cognite/client/_api/labels.py

```diff
@@ -156,3 +156,37 @@ def delete(self, external_id: str | SequenceNotStr[str] | None = None) -> None:
                 >>> client.labels.delete(external_id=["big_pump", "small_pump"])
         """
         self._delete_multiple(identifiers=IdentifierSequence.load(external_ids=external_id), wrap_ids=True)
+
+    @overload
+    def retrieve(self, external_id: str, ignore_unknown_ids: bool = False) -> None | LabelDefinition: ...
```

Comment:
`ignore_unknown_ids` doesn't make sense for the single-id case. This will never return `None` now. It should be removed from this signature and if `isinstance(external_id, str)` you can invoke `.as_singleton()` on the `IdentifierSequence` which will make the logic correct. See e.g. `assets.retrieve()`.

Reply:
let's add a test for the cases with unknown ids too.